pkg/kubelet/kubelet.go

```diff
@@ -1899,10 +1893,40 @@ func (kl *Kubelet) PLEGHealthCheck() (bool, error) {
 // and returns an error if the status check fails.  If the status check is OK,
 // update the container runtime uptime in the kubelet runtimeState.
 func (kl *Kubelet) updateRuntimeUp() {
-	if err := kl.containerRuntime.Status(); err != nil {
+	s, err := kl.containerRuntime.Status()
+	if err != nil {
 		glog.Errorf("Container runtime sanity check failed: %v", err)
 		return
 	}
+	// Only check specific conditions when runtime integration type is cri,
+	// because the old integration doesn't populate any runtime condition.
+	if kl.kubeletConfiguration.ExperimentalRuntimeIntegrationType == "cri" {
```

Comment:
Never mind. I decided to send in #36201 to enforce that all integration using CRI should specify --experimental-runtime-integration-type=cri for now to simplify the code.